dot/parachain/provisioner/prioritized_disputes_selection.go

```diff
@@ -8,18 +8,135 @@ import (
 	"fmt"
 	"time"
 
-	"github.com/ChainSafe/gossamer/dot/parachain/disputes-coordinator/messages"
+	disputemessages "github.com/ChainSafe/gossamer/dot/parachain/disputes-coordinator/messages"
 	parachain "github.com/ChainSafe/gossamer/dot/parachain/runtime"
 	parachaintypes "github.com/ChainSafe/gossamer/dot/parachain/types"
 	"github.com/ChainSafe/gossamer/lib/common"
+	"github.com/ChainSafe/gossamer/lib/primitives"
 )
 
 type BlockState interface {
 	GetRuntime(blockHash common.Hash) (instance parachain.RuntimeInstance, err error)
 }
 
-// IsVoteWorthToKeep determines if a vote is worth to be kept, based on the onchain disputes.
-func IsVoteWorthToKeep(
+// partitionedDisputes contains disputes by partitions.
+type partitionedDisputes struct {
+	// Concluded and inactive disputes which are completely unknown for the Runtime.
+	// Hopefully this should never happen. Will be sent to the Runtime with FIRST priority.
+	inactiveUnknownOnchain []parachaintypes.DisputeKey
+
+	// Disputes which are INACTIVE locally but they are unconcluded for the Runtime.
+	// A dispute can have enough local vote to conclude and at the same time the
+	// Runtime knows nothing about them at treats it as unconcluded. This discrepancy
+	// should be treated with high priority.
+	// Will be sent to the Runtime with SECOND priority.
+	inactiveUnconcludedOnchain []parachaintypes.DisputeKey
+
+	// Active disputes completely unknown onchain.
+	// Will be sent to the Runtime with THIRD priority.
+	activeUnknownOnchain []parachaintypes.DisputeKey
+
+	// Active disputes unconcluded onchain.
+	// Will be sent to the Runtime with FOURTH priority.
+	activeUnconcludedOnchain []parachaintypes.DisputeKey
+
+	// Active disputes concluded onchain. New votes are not that important for this partition.
+	// Will be sent to the Runtime with FIFTH priority.
+	activeConcludedOnchain []parachaintypes.DisputeKey
+
+	// Inactive disputes which has concluded onchain. These are not interesting and
+	// won't be sent to the Runtime.
+	// Will be DROPPED
+	inactiveConcludedOnchain []parachaintypes.DisputeKey
+}
+
+// orderedPartitions returns an array of partitions in the order they should be processed.
+func (pd partitionedDisputes) orderedPartitions() []parachaintypes.DisputeKey {
+	seqToIterate := [][]parachaintypes.DisputeKey{
+		pd.inactiveUnknownOnchain,
+		pd.inactiveUnconcludedOnchain,
+		pd.activeUnknownOnchain,
+		pd.activeUnconcludedOnchain,
+		pd.activeConcludedOnchain,
+		// pd.InactiveConcludedOnchain is dropped on purpose
```

Comment:
what purpose?

Reply:
they are out since the values they hold are not relevant for the runtime given that the disputes is already inactive (already closed) and is concluded on chain, so we can ignore any votes on that "partition"